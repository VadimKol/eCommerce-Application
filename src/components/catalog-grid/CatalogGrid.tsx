import styles from './styles.module.scss';

const sourceArr = [
  {
    title: 'Weapons',
    image: '/src/assets/images/home_weapon.jpg',
    link: '/catalog/cosplay/weapon',
  },
  {
    title: 'Wigs',
    image: '/src/assets/images/home_wig.jpg',
    link: '/catalog/cosplay/wigs',
  },
  {
    title: 'Mugs',
    image: '/src/assets/images/home_mug.png',
    link: '/catalog/decor/mugs',
  },
  {
    title: 'Pads',
    image: '/src/assets/images/home_pad.jpg',
    link: '/catalog/decor/pads',
  },
  {
    title: 'Notebooks',
    image: '/src/assets/images/home_note.jpg',
    link: '/catalog/stationery/notepads',
  },
];

export function CatalogGrid(): JSX.Element {
  return (
    <div className={styles.catalogBlock}>
      {sourceArr.map((item) => (
        <div key={item.title} className={styles.catalogItem}>
          <a href={item.link}>
            <img src={item.image} alt={item.title} className={styles.catalogImage} />
            <h3 className={styles.catalogTitle}>{item.title}</h3>
          </a>
        </div>
      ))}
    </div>
  );
}

export default CatalogGrid;
