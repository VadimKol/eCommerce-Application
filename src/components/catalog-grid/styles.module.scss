.catalogBlock {
  display: flex;
  flex-flow: row wrap;
  gap: 1rem;
  justify-content: space-between;
  align-items: center;
  width: 80%;
  margin: 50px auto 0;
}

.catalogItem {
  width: 30%;
  max-width: 450px;
  height: 350px;
  padding: 1rem;
  text-align: center;
  background: linear-gradient(var(--background-color) 0 0) padding-box, linear-gradient(45deg, var(--primary-color), var(--secondary-color)) border-box;
  border: 2px solid transparent;
  border-radius: 10px;

  & img {
    width: 250px;
    height: 250px;
    transition: all 500ms;
    transform: scale(1);
  }

  &:hover img {
    transition: all 500ms;
    transform: scale(1.1);
  }

  & a {
    display: flex;
    flex-flow: column nowrap;
    gap: 10px;
    justify-content: space-around;
    align-items: center;
    text-decoration: none;
  }
}

.catalogImage {
  max-width: 100%;
  height: auto;
}

.catalogTitle {
  margin-top: 0.5rem;
  font-size: 20px;
  text-decoration: none;
  color: var(--primary-color);
}

.catalogItem.catalogItemMore {
  width: 100%;
  max-width: 100%;
  height: 80px;
  transition: all 500ms;

  &:hover {
    box-shadow: -5px 5px 5px var(--secondary-color);
    transform: translateY(-2px);
  }
}

@media (width <= 1600px) {
  .catalogItem {
    width: 30%;
    height: 300px;

    & img {
      width: 200px;
      height: 200px;
    }
  }
}


@media (width <= 1400px) {
  .catalogItem {
    width: 45%;
    height: 200px;

    & img {
      width: 150px;
      height: 150px;
    }
  }
}

@media (width <= 900px) {
  .catalogItem {
    height: 220px;
  }

  .catalogItem a {
    flex-flow: column nowrap;
  }
}

@media (width <= 650px) {
  .catalogItem {
    width: 100%;
  }

  .catalogItem a {
    flex-flow: row nowrap;
  }
}
