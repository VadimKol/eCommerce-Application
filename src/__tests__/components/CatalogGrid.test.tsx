import { render } from '@testing-library/react';

import { CatalogGrid } from '@/components/catalog-grid/CatalogGrid';

it('renders correctly', () => {
  const { container } = render(<CatalogGrid />);
  expect(container).toMatchSnapshot();
});
