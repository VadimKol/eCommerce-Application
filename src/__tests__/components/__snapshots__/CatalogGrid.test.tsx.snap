// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="catalogBlock"
  >
    <div
      class="catalogItem"
    >
      <a
        class="link"
        href="/catalog/cosplay/weapon"
      >
        <img
          alt="Weapons"
          class="catalogImage"
          src="stub-test"
        />
        <h3
          class="catalogTitle"
        >
          Weapons
        </h3>
      </a>
    </div>
    <div
      class="catalogItem"
    >
      <a
        class="link"
        href="/catalog/cosplay/wigs"
      >
        <img
          alt="Wigs"
          class="catalogImage"
          src="stub-test"
        />
        <h3
          class="catalogTitle"
        >
          Wigs
        </h3>
      </a>
    </div>
    <div
      class="catalogItem"
    >
      <a
        class="link"
        href="/catalog/decor/figures"
      >
        <img
          alt="Figures"
          class="catalogImage"
          src="stub-test"
        />
        <h3
          class="catalogTitle"
        >
          Figures
        </h3>
      </a>
    </div>
    <div
      class="catalogItem"
    >
      <a
        class="link"
        href="/catalog/decor/mugs"
      >
        <img
          alt="Mugs"
          class="catalogImage"
          src="stub-test"
        />
        <h3
          class="catalogTitle"
        >
          Mugs
        </h3>
      </a>
    </div>
    <div
      class="catalogItem"
    >
      <a
        class="link"
        href="/catalog/decor/pads"
      >
        <img
          alt="Pads"
          class="catalogImage"
          src="stub-test"
        />
        <h3
          class="catalogTitle"
        >
          Pads
        </h3>
      </a>
    </div>
    <div
      class="catalogItem"
    >
      <a
        class="link"
        href="/catalog/stationery/notepads"
      >
        <img
          alt="Notepads"
          class="catalogImage"
          src="stub-test"
        />
        <h3
          class="catalogTitle"
        >
          Notepads
        </h3>
      </a>
    </div>
    <div
      class="catalogItem catalogItemMore"
    >
      <a
        class="link"
        href="/catalog"
      >
        <h3
          class="catalogTitle"
        >
          Show more at Catalog
        </h3>
      </a>
    </div>
  </div>
</div>
`;
